src/nbl/video/COpenGLCommon.h

```diff
@@ -256,14 +256,14 @@ inline GLenum	getSizedOpenGLFormatFromOurFormat(IOpenGL_FunctionTable* gl, asset
 			return GL_R8;
 			break;
 		case EF_R8_SRGB:
-			if (!gl->getFeatures()->isFeatureAvailable(COpenGLFeatureMap::NBL_EXT_texture_sRGB_R8))
+			if (gl && !gl->getFeatures()->isFeatureAvailable(COpenGLFeatureMap::NBL_EXT_texture_sRGB_R8))
```

Comment:
you can assume `gl` not null

Reply:
Actually I pass it `nullptr` here
https://github.com/Devsh-Graphics-Programming/Nabla/blob/vulkan_backend/src/nbl/video/COpenGLESPhysicalDevice.h#L73
Then should format reporting functions take in a function table?

Reply:
this gets tricky and sticky.

You need to query and cache all the formats up-front (or cache the `gl->getFeatures()`) because in order to do any OpenGL-calls you need a context current to the thread, and its out policy to never have any OpenGL contexts current in non-dedicated threads.

So either make it so you dont need to do any OpenGL calls to figure this out OR query all this stuff on the temporary context which is made during creation/construction to gather info about Sfeatures and SLimits (OpenGL physical device does not and never has a context live during its existence)

Reply:
@achalpandeyy status? If its not done, then tag Erfan in the convo

Reply:
Nope not done

CC: @Erfan-Ahmadi

Reply:
> Actually I pass it `nullptr` here https://github.com/Devsh-Graphics-Programming/Nabla/blob/vulkan_backend/src/nbl/video/COpenGLESPhysicalDevice.h#L73 Then should format reporting functions take in a function table?

Here the `peglGetProcAddress` call and function pointer to `glGetInternalformativ` should be cached in the function table like all other GL functions.

> query all this stuff on the temporary context which is made during creation/construction to gather info about Sfeatures and SLimits (OpenGL physical device does not and never has a context live during its existence)

Do this at context creation.

Between here:
https://github.com/Devsh-Graphics-Programming/Nabla/blob/baw/src/nbl/video/IOpenGL_PhysicalDeviceBase.h#L187
and here:
https://github.com/Devsh-Graphics-Programming/Nabla/blob/baw/src/nbl/video/IOpenGL_PhysicalDeviceBase.h#L523

you can do all your queries of OpenGL caps